libdnf/rpm/package.cpp

```diff
@@ -285,13 +285,16 @@ unsigned long long Package::get_rpmdbid() const {
     return solv::get_rpmdbid(pool, id);
 }
 
-Repo * Package::get_repo() const {
-    Pool * pool = sack->p_impl->pool;
-    return solv::get_repo(pool, id);
+libdnf::rpm::RepoWeakPtr Package::get_repo() const {
+    auto & repo_sack = sack->p_impl->base->get_rpm_repo_sack();
+    auto repo_query = repo_sack.new_query();
+    repo_query.ifilter_id(sack::QueryCmp::EQ, {get_repo_id()});
+    return repo_query.get();
```

Comment:
I assume you do this to get a proper `PepoWeakPtr`. In addition to this being roundabout and not very nice, the query creates a vector of weak pointers to all the repos and then erases them one by one until only the one you want is left. I think this needs some improvements :grin: But they'll likely need to be of larger scale.

Lets not inflate this PR, I'm ok with merging this (or leaving the previous implementation), but please add a TODO?

Reply:
I've replaced query with constructing a weak pointer.